File: lib_stater/src/main/java/com/sum/stater/utils/LaunchTimer.java
package com.sum.stater.utils;

import android.util.Log;

public class LaunchTimer {

    private static long sTime;

    public static void startRecord() {
        sTime = System.currentTimeMillis();
    }

    public static void endRecord() {
        endRecord("");
    }

    public static void endRecord(String msg) {
        long cost = System.currentTimeMillis() - sTime;
        Log.i("LaunchTimer", msg + "cost " + cost);
    }

}

File: lib_stater/src/main/java/com/sum/stater/utils/LaunchTimer.kt
package com.sum.stater.utils

import com.sum.framework.log.LogUtil

/**
 * 自定义埋点工具
 */
object LaunchTimer {
    object LaunchTimer {
        private var currentTime: Long = 0

        // 记录开始时间
        fun startRecord() {
            currentTime = System.currentTimeMillis()
        }

        // 记录结束时间，某个tag的耗时
        @JvmOverloads
        fun stopRecord(title: String = "") {
            val t = System.currentTimeMillis() - currentTime
            LogUtil.d("$title | time：$t", tag = "LaunchTimer")
        }
    }
}